src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509Encoder.cs

```diff
@@ -174,7 +174,10 @@ public unsafe void DecodeX509KeyUsageExtension(byte[] encoded, out X509KeyUsageF
                 if (data->pathlen != IntPtr.Zero)
                 {
                     hasPathLengthConstraint = true;
-                    pathLengthConstraint = Interop.libcrypto.ASN1_INTEGER_get(data->pathlen).ToInt32();
+                    long pathLengthConstraint64 = Interop.Crypto.Asn1IntegerGet(data->pathlen);
+
+                    Debug.Assert(pathLengthConstraint64 < int.MaxValue, "pathLengthConstraint needs to be in the Int32 range.");
```

Comment:
Since ASN1_INTEGER_get returns a 'long' and the shim is a direct shim, it feels more appropriate for the shim to return an Int64.  In this case, "pathlen" is pretty much guaranteed to fit inside an Int32, which makes the assert and cast valid, IMO.

The cast will throw an exception right? It wouldn't be the best exception possible, but it still would fail if there was ever a case where path length could be greater than 2 billion.